tests/firewall/tests.c

```diff
@@ -690,12 +690,51 @@ static void test_firewall_noip(void)
 	firewall_noip(EAST_SIDE);
 }
 
+static void test_firewall_rules(void)
+{
+	struct pg_error *error = NULL;
+	struct pg_brick *fw = pg_firewall_new("fw", 1, 1, PG_NONE, &error);
+	const char *r[] = {
+		"src host 10.0.0.1",
+		"udp",
+		"tcp",
+		"icmp",
+		"icmp or udp or tcp",
+		"udp dst portrange 4000-6000",
+		"tcp dst portrange 0-65535",
+		"tcp dst port 42",
+		"  ",
+		"",
+		"src host 10.0.0.1 and (icmp or udp or tcp)",
+		"((src net 0.0.0.0/0 and udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 && udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 or udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 || udp dst portrange 4000-6000))",
+		NULL
+	};
+
+	for (int i = 0; r[i]; i++) {
+		if (pg_firewall_rule_add(fw, r[i], WEST_SIDE, 0, &error) < 0) {
```

Comment:
I'm not really sure of the utility of this if, you could simply do a
``` g_assert(!pg_firewall_rule_add(fw, r[i], WEST_SIDE, 0, &error)) ```, you would have the same result except you will not print the error, but you should not print error in test